src/token/erc6909/extensions/erc6909_metadata.cairo

```diff
@@ -0,0 +1,134 @@
+/// The internal function `_update_token_metadata` should be used inside the ERC6909 Hooks.
+#[starknet::component]
+pub mod ERC6909MetadataComponent {
+    use core::num::traits::Zero;
+    use openzeppelin::token::erc6909::ERC6909Component;
+    use openzeppelin::token::erc6909::interface;
+    use starknet::ContractAddress;
+
+    #[storage]
+    struct Storage {
+        ERC6909Metadata_name: LegacyMap<u256, ByteArray>,
+        ERC6909Metadata_symbol: LegacyMap<u256, ByteArray>,
+        ERC6909Metadata_decimals: LegacyMap<u256, u8>,
+    }
+
+    #[embeddable_as(ERC6909MetadataImpl)]
+    impl ERC6909Metadata<
+        TContractState,
+        +HasComponent<TContractState>,
+        +ERC6909Component::HasComponent<TContractState>,
+        +ERC6909Component::ERC6909HooksTrait<TContractState>,
+        +Drop<TContractState>
+    > of interface::IERC6909Metadata<ComponentState<TContractState>> {
+        /// @notice Name of a given token.
+        /// @param id The id of the token.
+        /// @return The name of the token.
+        fn name(self: @ComponentState<TContractState>, id: u256) -> ByteArray {
+            self.ERC6909Metadata_name.read(id)
+        }
+
+        /// @notice Symbol of a given token.
+        /// @param id The id of the token.
+        /// @return The symbol of the token.
+        fn symbol(self: @ComponentState<TContractState>, id: u256) -> ByteArray {
+            self.ERC6909Metadata_symbol.read(id)
+        }
+
+        /// @notice Decimals of a given token.
+        /// @param id The id of the token.
+        /// @return The decimals of the token.
+        fn decimals(self: @ComponentState<TContractState>, id: u256) -> u8 {
+            self.ERC6909Metadata_decimals.read(id)
+        }
+    }
+
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC6909: ERC6909Component::HasComponent<TContractState>,
+        +ERC6909Component::ERC6909HooksTrait<TContractState>,
+        +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// @notice Updates the metadata of a token ID.
+        /// @notice Ideally this function should be called in a `before_update` or `after_update` hook during mints.
+        /// @param sender The address of the sender.
+        /// @param id The ID of the token.
+        /// @param name The name of the token.
+        /// @param symbol The symbol of the token.
+        /// @param decimals The decimals of the token.
+        fn _update_token_metadata(
+            ref self: ComponentState<TContractState>,
+            sender: ContractAddress,
+            id: u256,
+            name: ByteArray,
+            symbol: ByteArray,
+            decimals: u8
+        ) {
+            let zero_address = Zero::zero();
+
+            // In case of new ID mints update the token metadata
+            if (sender == zero_address) {
```

Comment:
```suggestion
            if (sender.is_zero()) {
```

Reply:
Fixed in [40538f6](https://github.com/OpenZeppelin/cairo-contracts/pull/1040/commits/40538f672aaef69bf8dc96d3c9a86b3d4c9abbb2) thank you!